File: Code/Websites/DanpheEMR/wwwroot/DanpheApp/src/app/settings-new/shared/price.category.model.ts
import { FormBuilder, FormGroup, Validators } from "@angular/forms";


export class PriceCategory {

  public PriceCategoryId: number = 0;
  public PriceCategoryCode: string = null;
  public PriceCategoryName: string = null;
  public Description: string = null;
  public ShowInRegistration: boolean = false;
  public ShowInAdmission: boolean = false;
  public IsDefault: boolean = false;
  public IsRateDifferent: boolean = false;

  public CreatedBy: number = 0;
  public CreatedOn: string = '';
  public IsActive: boolean = true;
  public PharmacyDefaultCreditOrganizationId: number = null;
  public IsPharmacyRateDifferent: boolean = false;
  public DisplaySequence: number = 0;


  /*Manipal-RevisionNeeded*/
  //Start: Sud:22Mar'23--Below Fields should be removed as per NewBillingStructure
  public IsCoPayment: boolean = false;
  public DefaultCreditOrganizationId: number = null;
  public Copayment_CashPercent: number = null;
  public Copayment_CreditPercent: number = null;
  //End: Sud:22Mar'23--Below Fields should be removed as per NewBillingStructure

  public PriceValidator: FormGroup;


  constructor() {
    var _formBuilder = new FormBuilder();
    this.PriceValidator = _formBuilder.group({
      'PriceCategoryName': ['', Validators.compose([Validators.required])],

    });
  }
  public IsDirty(fieldName): boolean {
    if (fieldName == undefined) {
      return this.PriceValidator.dirty;
    }

    else
      return this.PriceValidator.controls[fieldName].dirty;
  }

  public IsValid(): boolean {
    if (this.PriceValidator.valid) {
      return true;
    }
    else {
      return false;
    }
  }

  public IsValidCheck(fieldName, validator): boolean {

    if (fieldName == undefined) {
      return this.PriceValidator.valid;
    }
    else {
      return !(this.PriceValidator.hasError(validator, fieldName));
    }

  }



}

File: Code/Components/DanpheEMR.ServerModel/BillingModels/Config/PriceCategoryModel.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace DanpheEMR.ServerModel
{
    public class PriceCategoryModel
    {
        [Key]
        public int PriceCategoryId { get; set; }
        public string PriceCategoryName { get; set; }
        public string Description { get; set; }
        public bool IsDefault { get; set; }
        public DateTime CreatedOn { get; set; }
        public int CreatedBy { get; set; }
        public bool IsActive { get; set; }
        public bool? IsPharmacyRateDifferent { get; set; }
        public string PriceCategoryCode { get; set; } //Krishna, 16thFeb'23 New Column need to handle it where needed
        public bool ShowInRegistration { get; set; }
        public bool ShowInAdmission { get; set; }
        public int? DisplaySequence { get; set; }
        public int? ModifiedBy { get; set; }
        public DateTime? ModifiedOn { get; set; }


        //Krishna, 16thFeb'23 Below columns are deprecated in the new structure, Hence this need impact handling, and can be removed after that
        //public bool? IsCoPayment { get; set; }
        //public decimal? Copayment_CashPercent { get; set; }
        //public decimal? Copayment_CreditPercent { get; set; }
        //public int? PharmacyDefaultCreditOrganizationId { get; set; }

    }


}
